sound/soc/sof/ipc4-topology.c

```diff
@@ -544,6 +545,7 @@ static int sof_ipc4_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 	kfree(available_fmt->dma_buffer_size);
 free_copier:
 	kfree(ipc4_copier);
+	dai->private = NULL;
```

Comment:
what about dai->scomp, it's also modified in this routine?

Reply:
> what about dai->scomp, it's also modified in this routine?

@plbossart We don't free scomp here. So it should be safe. But yes, it should be better to handle it here. I will set `dai->scomp = NULL` in the error handling.